### PR TITLE
Add folder inclusion and exclusion for content validator

### LINKED ISSUE #373: Provide ability to filter content validator to include or exclude specific folders
Currently, we let users exclude errors from personal folders, but they can't get more granular than that. We should allow users to (1) run the content validator only on specific folders and (2) exclude specific folders from the content validator.

**Acceptance Tests**

- [ ] Can pass `--folders` with a comma separated list of integers. This would filter the content validation results for those folders
- [ ] Can pass `--exclude-folders` with a comma separated list of integers. This would filter out any content validation results for those folders.

### PR DESCRIPTION
**Pull request by DylanBaker**
## Change description

This PR is a replacement for PR #376. It is being re-created so that tests can be run in CircleCI. We haven't yet got things configured properly for forked branches to be run. 

This PR adds the ability to run the content validator on specific folders or to exclude specific folders from the result. It adds two new arguments: `--folders` and `--exclude-folders`. The exclusion list will take precedence over the inclusion list, i.e. if folder `1` is both included and excluded, it won't be selected. 

## Type of change
- [ ] Bug fix (fixes an issue)
- [x] New feature (adds functionality)

## Related issues

Closes #373 

### Security

- [X] Security impact of change has been considered
- [X] Code follows security best practices and guidelines

### Code review 

- [X] Pull request has a descriptive title and context useful to a reviewer

### CODE REVIEW

File: spectacles/validators/content.py [hunk 1]
```diff
@@ -22,6 +32,31 @@ def _get_personal_folders(self) -> List[int]:
                 personal_folders.append(folder["id"])
         return personal_folders
 
+    def _get_all_subfolders(self, input_folders: List[int]) -> List[int]:
+        result = []
+        all_folders = self.client.all_folders(self.project.name)
```
Comment (joshtemple) [inline spectacles/validators/content.py:37]: Looks like we still call this twice. Are you okay with that?

File: tests/test_content_validator.py [hunk 2]
```diff
@@ -83,3 +83,79 @@ def test_personal_folder_content_should_not_be_present(self, validator_fail):
...
+            yield validator, results
+
+    def test_personal_folder_content_should_not_be_present(
+        self, validator_fail_with_exclude
+    ):
+        results = validator_fail_with_exclude[1]
+        assert len(results["errors"]) == 0
+
+
+class TestValidateFailIncludeFolder:
```
Comment (joshtemple) [inline tests/test_content_validator.py:113]: Can you add a test for what happens when the user specifies a folder ID that doesn't exist?

Comment (DylanBaker) [reply tests/test_content_validator.py:113]: Yes, I'll add that. Good call.

File: spectacles/validators/content.py (hunk 1 above)
Comment (DylanBaker) [reply spectacles/validators/content.py:37]: I think I am. It's now once for personal folders and once for subfolders, which I think I'm happy with. Would you be inclined to consolidate further?

Comment (joshtemple) [reply spectacles/validators/content.py:37]: Only if it's straightforward to do so. I don't think 1 extra call is going to have much of a performance hit.

**Review by joshtemple — APPROVED** [review]
LGTM!

**Pull request merged**